constrained-voucher.mkd

```diff
@@ -662,6 +669,38 @@ If the certificate that the Registrar uses is marked as a id-kp-cmcRA certificat
 
 In summary for typical Registrar use, where a single Registrar certificate is used, then the certificate MUST have EKU of: id-kp-cmcRA, id-kp-serverAuth, id-kp-clientAuth.
 
+## MASA Extensions {#brski-masa-extensions}
+
+This document supports efficient operation of Pledges in constrained networks, by letting a Pledge optionally omit the sending of its root CA certificate in the DTLS handshake with the Registrar.
+This optimization requires a Registrar to have an alternative method to obtain this root CA certificate, which it may need for validation and logging purposes. The alternative method defined here is 
+that the Registrar can request the relevant root CA certificate(s) directly from the MASA. Although the operator of a Registrar may have already obtained such CA certificate(s) via a vendor-specific method 
+(e.g. part of sales integration), a standardized method is still needed for cases where such vendor-specific methods do not exist or are not known to the operator.
+
+### Pledge CA Certificates Request
+A Registrar, after having established a TLS connection to the MASA, MAY request the full set of root CA certificates for all Pledges that are managed in scope of this MASA, by sending a Pledge CA 
+Certificates Request. This uses a HTTPS GET with the operation MASA URI path of "/.well-known/brski/cacerts".
+
+### Pledge CA Certificates Response
+A MASA server MUST support the HTTPS resource "/.well-known/brski/cacerts" for the GET method over the TLS connection where the client is a Registrar.
+
+If the request is processed successfully, the server response MUST have an HTTP 200 response code.  Any other response code indicates an error and the client MAY either retry the operation 
+or (eventually) abort the protocol.
+
+A successful response MUST be a certs-only CMC Simple PKI Response,
+as defined in {{RFC5272}}, containing the certificates described in the
+following paragraph.  The HTTP content-type of "application/pkcs7-mime" is used. The Simple PKI Response is sent
+with a Content-Transfer-Encoding of "base64" {{RFC2045}}.
```

Comment:
Didn't we create an alternative for certs-only?
Wouldn't it be better to mandate that?

Reply:
If we are are returning an SMIME document, maybe we should put some content in it for the human to review.

Reply:
The only alternative we have finished was for the constrained Pledge: Coap content format 287 per RFC 9148. This is "application/pkix-cert" and can carry only a single X509 certificate. This can work well if the Registrar would signal the exact identity of the Pledge to the MASA and then the MASA can return the correct root CA cert for it.

That would also avoid the problem of MASA possibly needing to return many (100s?) of certificates to the Registrar , not knowing which one the Registrar is going to need.

(We did consider some alternatives for fetching multiple certs, e.g. via Coap resources name/0, name/1, etc. to fetch N certs one by one. But not pursued)